<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>675</width>
    <height>674</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenInverter FOC Simulator</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>10</y>
      <width>171</width>
      <height>421</height>
     </rect>
    </property>
    <property name="title">
     <string>Vehicle Parameters</string>
    </property>
    <widget class="QWidget" name="formLayoutWidget">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>171</width>
       <height>401</height>
      </rect>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="labelVehicleWeight">
        <property name="text">
         <string>Vehicle Weight (kg)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="vehicleWeight">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>300</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelWheelSize">
        <property name="text">
         <string>Wheel Radius (m)</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="wheelSize">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>0.3</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="labelGearRatio">
        <property name="text">
         <string>Gear Ratio</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="gearRatio">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>6</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="labelVdc">
        <property name="text">
         <string>Vdc (V)</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="Vdc">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>350</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="labelLq">
        <property name="text">
         <string>Lq (mH)</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLineEdit" name="Lq">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>25</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="labelLd">
        <property name="text">
         <string>Ld (mH)</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="labelRs">
        <property name="text">
         <string>Rs (Ohm)</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QLineEdit" name="Rs">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>0.075</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLineEdit" name="Ld">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>25</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QLineEdit" name="FluxLinkage">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>90</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="labelFluxLink">
        <property name="text">
         <string>λ ( mWeber)</string>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QLineEdit" name="FluxLinkageDelta">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>0.3</string>
        </property>
       </widget>
      </item>
      <item row="9" column="1">
       <widget class="QLineEdit" name="SyncDelay">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>16</string>
        </property>
       </widget>
      </item>
      <item row="10" column="1">
       <widget class="QLineEdit" name="Drag">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>100</string>
        </property>
       </widget>
      </item>
      <item row="11" column="1">
       <widget class="QLineEdit" name="LoopFreq">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>8333</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="labelFluxLinkDelta">
        <property name="text">
         <string>λdelta ( %/AId)</string>
        </property>
       </widget>
      </item>
      <item row="9" column="0">
       <widget class="QLabel" name="labelSyncDelay">
        <property name="text">
         <string>SyncDel (us)</string>
        </property>
       </widget>
      </item>
      <item row="10" column="0">
       <widget class="QLabel" name="labelDrag">
        <property name="text">
         <string>Drag (%)</string>
        </property>
       </widget>
      </item>
      <item row="11" column="0">
       <widget class="QLabel" name="labelLoopFreq">
        <property name="text">
         <string>Loop Frequency (Hz)</string>
        </property>
       </widget>
      </item>
      <item row="12" column="0">
       <widget class="QLabel" name="labelSampPoint">
        <property name="text">
         <string>Sampling Point (%)</string>
        </property>
       </widget>
      </item>
      <item row="12" column="1">
       <widget class="QLineEdit" name="SamplingPoint">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>100</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>10</y>
      <width>171</width>
      <height>281</height>
     </rect>
    </property>
    <property name="title">
     <string>Run Controls</string>
    </property>
    <widget class="QWidget" name="formLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>171</width>
       <height>261</height>
      </rect>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0">
       <widget class="QPushButton" name="pbRunFor">
        <property name="text">
         <string>Run For (s)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="runTime">
        <property name="text">
         <string>5</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="pbRunFor10s">
        <property name="text">
         <string>Run (10s)</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QPushButton" name="pbRunFor1s">
        <property name="text">
         <string>Run (1s)</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QPushButton" name="pbRunFor100ms">
        <property name="text">
         <string>Run (100ms)</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QPushButton" name="pbRunFor10ms">
        <property name="text">
         <string>Run (10ms)</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QPushButton" name="pbStep">
        <property name="text">
         <string>Step</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>80</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="7" column="0">
       <widget class="QPushButton" name="pbRestart">
        <property name="text">
         <string>Restart</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>460</y>
      <width>171</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>Flux Linkage Calculator</string>
    </property>
    <widget class="QWidget" name="formLayoutWidget_3">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>171</width>
       <height>112</height>
      </rect>
     </property>
     <layout class="QFormLayout" name="formLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="labelBaseFreq">
        <property name="text">
         <string>Base Frequency (rpm)</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelBaseVolts">
        <property name="text">
         <string>Base Voltage (V)</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="BaseVolts">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>300</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="labelCalcFluxLink">
        <property name="text">
         <string>λ ( Weber)</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="CalcFluxLink">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>0.14</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="BaseFreq">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>5000</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>10</y>
      <width>201</width>
      <height>591</height>
     </rect>
    </property>
    <property name="title">
     <string>OpenInverter Parameters</string>
    </property>
    <widget class="QWidget" name="formLayoutWidget_4">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>201</width>
       <height>571</height>
      </rect>
     </property>
     <layout class="QFormLayout" name="formLayout_4">
      <item row="0" column="0">
       <widget class="QLabel" name="labelTorqueDemand">
        <property name="text">
         <string>Torque Demand (%)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="torqueDemand">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelThrottlCurrent">
        <property name="text">
         <string>Throttle Current (A/%)</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="throttleCurrent">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>3</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="labelOpMode">
        <property name="text">
         <string>Mode (1=Run,2=Man)</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="opMode">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>1</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="labelDir">
        <property name="text">
         <string>Direction (-1=R,0=N,1=F)</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="direction">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>1</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="labelIqManual">
        <property name="text">
         <string>Iq Manual</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLineEdit" name="IqManual">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="labelIdManual">
        <property name="text">
         <string>Id Manual</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLineEdit" name="IdManual">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>80</width>
          <height>10</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="labelPoles">
        <property name="text">
         <string>Motor Poles</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QLineEdit" name="Poles">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>4</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="labelCurKp">
        <property name="text">
         <string>Current Kp</string>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QLineEdit" name="CurrentKp">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>2000</string>
        </property>
       </widget>
      </item>
      <item row="9" column="0">
       <widget class="QLabel" name="labelCurKi">
        <property name="text">
         <string>Current Ki</string>
        </property>
       </widget>
      </item>
      <item row="9" column="1">
       <widget class="QLineEdit" name="CurrentKi">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>800</string>
        </property>
       </widget>
      </item>
      <item row="10" column="0">
       <widget class="QLabel" name="labelFWKp">
        <property name="text">
         <string>FW Kp</string>
        </property>
       </widget>
      </item>
      <item row="10" column="1">
       <widget class="QLineEdit" name="FWKp">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>32</string>
        </property>
       </widget>
      </item>
      <item row="11" column="0">
       <widget class="QLabel" name="labelFWKi">
        <property name="text">
         <string>FW Ki</string>
        </property>
       </widget>
      </item>
      <item row="11" column="1">
       <widget class="QLineEdit" name="FWKi">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>40</string>
        </property>
       </widget>
      </item>
      <item row="12" column="0">
       <widget class="QLabel" name="labelVLimKp">
        <property name="text">
         <string>ULim Kp</string>
        </property>
       </widget>
      </item>
      <item row="12" column="1">
       <widget class="QLineEdit" name="VLimKp">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>3</string>
        </property>
       </widget>
      </item>
      <item row="13" column="0">
       <widget class="QLabel" name="labelVLimKi">
        <property name="text">
         <string>ULim Ki</string>
        </property>
       </widget>
      </item>
      <item row="13" column="1">
       <widget class="QLineEdit" name="VLimKi">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>10</string>
        </property>
       </widget>
      </item>
      <item row="14" column="0">
       <widget class="QLabel" name="labelLqMinusLd">
        <property name="text">
         <string>Lq - Ld (mH)</string>
        </property>
       </widget>
      </item>
      <item row="14" column="1">
       <widget class="QLineEdit" name="LqMinusLd">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="15" column="0">
       <widget class="QLabel" name="labelSyncAdv">
        <property name="text">
         <string>SyncAdv (dig/Hz)</string>
        </property>
       </widget>
      </item>
      <item row="15" column="1">
       <widget class="QLineEdit" name="SyncAdv">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="16" column="0">
       <widget class="QLabel" name="labelFWStart">
        <property name="text">
         <string>FW Start (Hz)</string>
        </property>
       </widget>
      </item>
      <item row="16" column="1">
       <widget class="QLineEdit" name="FWStart">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>83</string>
        </property>
       </widget>
      </item>
      <item row="17" column="0">
       <widget class="QLabel" name="labelFMax">
        <property name="text">
         <string>FMax (Hz)</string>
        </property>
       </widget>
      </item>
      <item row="17" column="1">
       <widget class="QLineEdit" name="FreqMax">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>166</string>
        </property>
       </widget>
      </item>
      <item row="18" column="0">
       <widget class="QCheckBox" name="ExtraCycleDelay">
        <property name="layoutDirection">
         <enum>Qt::RightToLeft</enum>
        </property>
        <property name="text">
         <string>Extra Cycle Delay</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>675</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
